frontend/admin/src/js/components/user/GeneralInformationTab/AddToPoolDialog.tsx

```diff
@@ -57,24 +63,68 @@ export const AddToPoolDialog: React.FC<AddToPoolDialogProps> = ({
   const submitForm: SubmitHandler<FormValues> = async (
     formValues: FormValues,
   ) => {
-    await requestMutation({
-      pool: {
-        connect: formValues.poolId,
-      },
-      user: {
-        connect: user.id,
-      },
-      expiryDate: formValues.expiryDate,
-    })
-      .then(() => {
-        toast.success(
-          intl.formatMessage({
-            defaultMessage: "User added successfully",
-            id: "O8U5Sz",
-            description:
-              "Toast for successful add user to pool on view-user page",
-          }),
+    const poolsToUpdate = formValues.pools
+      .map((poolId) => poolMap.get(poolId))
+      .filter(notEmpty);
+
+    const promises = poolsToUpdate.map((pool) => {
+      return requestMutation({
+        pool: {
+          connect: pool?.id,
+        },
+        user: {
+          connect: user.id,
+        },
+        expiryDate: formValues.expiryDate,
+      }).catch((err) => {
+        throw err;
+      });
+    });
+
+    Promise.allSettled(promises)
+      .then((settledResults) => {
+        const requestsWithResults = zipWith(
+          poolsToUpdate,
+          settledResults,
+          (pool, settledResult) => {
+            return {
+              pool,
+              settledResult,
+            };
+          },
         );
+        const rejectedRequests = requestsWithResults.filter(
+          (r) => r.settledResult.status === "rejected",
+        );
+
+        if (!rejectedRequests.length) {
+          toast.success(
+            intl.formatMessage({
+              defaultMessage: "User added successfully",
+              id: "O8U5Sz",
+              description:
+                "Toast for successful add user to pool on view-user page",
+            }),
+          );
+        } else {
+          toast.error(
+            <>
+              {intl.formatMessage({
+                defaultMessage: "Failed to add user to the following pools:",
+                id: "6Y6hy9",
```

Comment:
do we have the french translation for this?

Reply:
I'm just waiting to get it back right now

Reply:
probably you could do it with the existing message like this 
https://github.com/GCTC-NTGC/gc-digital-talent/pull/5295/files#diff-15a2f6f5b8b70c1bf926290de4dce397d0939765ca1959181f96c8a4b40cff24R131.


I believe the existing msg for this 
```
toast.error(
          intl.formatMessage({
            defaultMessage: "Failed adding user",
            id: "GZqEuI",
            description: "Toast for failed add user to pool on view-user page",
          }),
```

Reply:
Added in [add translation](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5256/commits/4db8d307d6603f80872be380c14695f06748b520)